sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

```diff
@@ -5,40 +5,45 @@
 
 import static com.azure.core.util.CoreUtils.isNullOrEmpty;
 
+import java.util.Objects;
+
 import com.azure.core.http.HttpPipelineCallContext;
 import com.azure.core.http.HttpPipelineNextPolicy;
 import com.azure.core.http.HttpRequest;
 import com.azure.core.http.HttpResponse;
-import java.util.Objects;
+
 import com.azure.core.util.logging.ClientLogger;
+
 import reactor.core.publisher.Mono;
 
 import java.time.Duration;
 
 /**
  * A pipeline policy that retries when a recoverable HTTP error occurs.
+ * @see RetryPolicyOptions
  */
 public class RetryPolicy implements HttpPipelinePolicy {
 
-    private static final String RETRY_AFTER_MS_HEADER = "retry-after-ms";
-
     private final ClientLogger logger = new ClientLogger(RetryPolicy.class);
-    private final RetryStrategy retryStrategy;
+
+    private final RetryPolicyOptions retryPolicyOptions;
 
     /**
-     * Creates a default {@link ExponentialBackoff} retry policy.
+     * Creates a default {@link RetryPolicy} with default {@link RetryPolicyOptions} for retry policy.
      */
     public RetryPolicy() {
-        this(new ExponentialBackoff());
+        this(new RetryPolicyOptions());
     }
 
     /**
-     * Creates a RetryPolicy with the provided {@link RetryStrategy}.
+     * Creates a {@link RetryPolicy} with the provided {@link RetryPolicyOptions}.
      *
-     * @param retryStrategy The {@link RetryStrategy} used for retries.
+     * @param retryPolicyOptions with given {@link RetryPolicyOptions}.
+     * @throws NullPointerException if {@code retryPolicyOptions} is {@code null}.
      */
-    public RetryPolicy(RetryStrategy retryStrategy) {
-        this.retryStrategy = Objects.requireNonNull(retryStrategy, "'retryStrategy' cannot be null");
+    public RetryPolicy(RetryPolicyOptions retryPolicyOptions) {
```

Comment:
This is an API breaking change.

Reply:
yes we have to change any Client library using this, If we are fine with overall approach for all the Policy to have following constructor 

RetryPolicy()
RetryPolicy(RetryOptions)

Reply:
Adding the constructor back.